presto-server-rpm/src/main/script/symlink.groovy

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import java.nio.file.Files
+import java.nio.file.LinkOption
+import java.nio.file.Path
+import java.nio.file.Paths
+
+Path root = Paths.get(properties['root'])
+Path shared = root.resolve('shared')
+
+Files.createDirectory(shared)
+
+List<String> files = new FileNameFinder().getFileNames(root.toString(), '**/*.jar')
+
+for (file in files) {
+    Path source = Paths.get(file)
+    Path target = shared.resolve(source.getFileName())
```

Comment:
Can we use content checksum here?

jar name = artifact + id, without group, so it's not unique

and this would _also_ prevent potential surprises like https://github.com/prestosql/presto/issues/2470

Reply:
I considered that, but the current hardlink handling in provisio only uses the name, so it doesn't buy us anything at this point. If we implement that in provisio, this should change, too.

Reply:
I'll add a comment to this effect here.

Reply:
I understand, but this is something we should address, so why not do right thing for RPM from the start?
I don't think it's difficult or expensive to do.

Reply:
You're right, it's easy to do. I added a check that the files are the same.

Reply:
Thanks